conjur/init_logic.py

```diff
@@ -0,0 +1,107 @@
+# Builtin
+import logging
+import sys
+import os.path
+from pathlib import Path
+
+# Third party
+import socket
+import yaml
+from OpenSSL import SSL
+
+class InitLogic:
+  """
+  InitLogic
+
+  This class holds the business logic for population the  conjurrc configuration details needed to connect with Conjur
+  """
+  DEFAULT_CONFIG_FILE = os.path.expanduser(os.path.join('~', '.conjurrc'))
+  DEFAULT_PROMPT_RESPONSE=True
+
+  def __init__(self):
+    pass
+
+  @staticmethod
+  def connect_to_socket(hostname, port):
+    """
+    Method for opening a socket with the Conjur server
+    """
+    try:
+      if port == None:
+        port = 443
+
+      ctx = SSL.Context(method=SSL.TLSv1_2_METHOD)
+      # We set the following to false because we want to fetch the certificates and not use them to validate yet
+      ctx.check_hostname = False
+      ctx.verify_mode = False
+
+      sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+      sock = SSL.Connection(context=ctx, socket=sock)
```

Comment:
You override the sock here maybe rename it to conjur_connection/conjur_sock/conjur_client